StrongInject.Generator/Visitors/LoweringVisitor.cs

```diff
@@ -151,7 +162,9 @@ protected override void AfterVisit(InstanceSource source, State state)
                     isSingleInstanceCreation: false,
                     isAsyncContext: delegateSource.IsAsync,
                     singleInstanceVariablesInScope: _existingVariables.Where(x => x.Key.Scope == Scope.SingleInstance),
+                    _currentlyVisitingDelegates,
                     targetName: out var targetName);
+                _currentlyVisitingDelegates.Remove(delegateSource);
```

Comment:
Reflexively I'd expect this to be in a `finally` block.

Reply:
All exceptions in stronginject are fatal - we never throw any exceptions as part of normal control flow, and never catch exceptions.